src/coreclr/vm/interpexec.cpp

```diff
@@ -1213,6 +1214,32 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     ip += 5;
                     goto CALL_INTERP_SLOT;
                 }
+                case INTOP_NEWOBJ_VAROBJSIZE:
+                {
+                    returnOffset = ip[1];
+                    callArgsOffset = ip[2];
+                    methodSlot = ip[3];
+
+                    // FIXME: Duplicated code from CALL_INTERP_SLOT
```

Comment:
I don't really understand why this opcode is not a normal call like the others. Could we avoid having this code duplication here ?

Reply:
generating the call and getting everything to work right given the way we do the tagged method pointer and then cache the call target looked like a pain.

right now we rely on being able to cache the call target and then use CodeInfo to figure out whether it is interp code or jit code. the helpers for this are a third category so we would need another different tag for them or would need to generate a generic helper opcode.

if we're not ok with a special opcode for arrays and strings i can figure something out, but i don't know how long it will take